Makefile

```diff
@@ -0,0 +1,160 @@
+ROOT_DIR := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+export LLC := llc
+export CLANG := clang
+export OBJCOPY := llvm-objcopy
+
+CARGO := cargo
+RELEASE_VERSION ?= $(shell $(CARGO) metadata --no-deps --format-version=1 | jq -r '.packages | .[] | select(.name == "retis") | .version')
+RELEASE_NAME ?= $(shell $(CARGO) metadata --no-deps --format-version=1 | jq '.packages | .[] | select(.name=="retis") | .metadata.misc.release_name')
+
+PRINT = echo
+
+VERBOSITY := $(filter 1,$(V))$(filter 2,$(V))
+
+ifeq ($(VERBOSITY),)
+    Q=@
+    MAKE += -s
+    CARGO += -q
+define out_console
+    $(PRINT) -e "[$(1)]\t$(2)"
+endef
+
+.SILENT:
+else
+    ifeq ($(VERBOSITY),2)
+        CARGO_CMD_OPTS += --verbose
+    endif
+endif
+
+ifeq ($(NOVENDOR),)
+    # This MUST be kept in sync with API_HEADERS under lib.rs in libbpf-sys
+    LIBBPF_API_HEADERS := bpf.h \
+                          libbpf.h \
+                          btf.h \
+                          bpf_helpers.h \
+                          bpf_helper_defs.h \
+                          bpf_tracing.h \
+                          bpf_endian.h \
+                          bpf_core_read.h \
+                          libbpf_common.h \
+                          usdt.bpf.h
+
+    LIBBPF_SYS_LIBBPF_BASE_PATH := $(dir $(shell cargo metadata --format-version=1 | jq -r '.packages | .[] | select(.name == "libbpf-sys") | .manifest_path'))
+    LIBBPF_SYS_LIBBPF_INCLUDES :=  $(wildcard $(addprefix $(LIBBPF_SYS_LIBBPF_BASE_PATH)/libbpf/src/, $(LIBBPF_API_HEADERS)))
+    LIBBPF_INCLUDES := $(ROOT_DIR)/src/.out
+endif
+
+FILTER_INCLUDES := src/core/filters/packets/bpf/include \
+                   src/core/filters/meta/bpf/include
+INCLUDES_ALL := $(abspath $(wildcard src/core/probe/bpf/include \
+                                     src/core/probe/kernel/bpf/include \
+                                     src/core/probe/user/bpf/include \
+                                     src/core/events/bpf/include \
+                                     src/core/tracking/bpf/include \
+                                     /usr/include/x86_64-linux-gnu \
+                                     $(FILTER_INCLUDES)))
+INCLUDES_ALL += $(LIBBPF_INCLUDES)
+
+OVS_INCLUDES := $(abspath src/module/ovs/bpf/include)
+INCLUDES := $(addprefix -I, $(INCLUDES_ALL))
+
+EBPF_PROBES := $(abspath src/core/probe/kernel/bpf \
+                        src/core/probe/user/bpf)
+
+GENERIC_HOOKS := $(abspath src/module/skb/bpf \
+                           src/module/skb_drop/bpf \
+                           src/module/skb_tracking/bpf \
+                           src/module/nft/bpf \
+                           src/module/ct/bpf)
+
+OVS_HOOKS := $(abspath src/module/ovs/bpf)
+OUT_NAME := HOOK
+
+JOBS := $(patsubst -j%,%,$(filter -j%,$(MAKEFLAGS)))
+
+ifneq ($(JOBS),)
+    CARGO_JOBS := $(JOBS)
+else
+    CARGO_JOBS := 1
+endif
+
+all: build
+
+install: release
+	$(CARGO) $(CARGO_OPTS) install $(CARGO_CMD_OPTS) --path=.
```

Comment:
I don't think we can use the same `CARGO_CMD_OPTS` that is used for building. We could special case the above (`CARGO_INSTALL_OPTS`) or remove it.

Reply:
makes sense